### ISSUE TITLE
Build failure: pkgsLLVM.xorg.libpciaccess

### ISSUE THREAD

**Issue by RossComputerGuy** [NixOS/nixpkgs#310726]
### Steps To Reproduce

1. build `pkgsLLVM.xorg.libpciaccess`

### Build log

https://gist.github.com/RossComputerGuy/80ac10b0a8adeeaba0ffe75bc0fbcdda

### Additional context

Found while compiling mesa with `pkgsLLVM`.

### Notify maintainers

@sdht0

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"aarch64-linux"`
 - host os: `Linux 5.15.156, NixOS, 24.05 (Uakari), 24.05.20240418.bc279bb`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - nixpkgs: `/run/current-system/nixpkgs`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by rrbutani** [comment]
```
FAILED: src/libpciaccess.so.0.11.1
aarch64-unknown-linux-gnu-clang  -o src/libpciaccess.so.0.11.1 src/libpciaccess.so.0.11.1.p/common_bridge.c.o src/libpciaccess.so.0.11.1.p/common_iterator.c.o src/libpciaccess.so.0.11.1.p/common_init.c.o src/libpciaccess.so.0.11.1.p/common_interface.c.o src/libpciaccess.so.0.11.1.p/common_io.c.o src/libpciaccess.so.0.11.1.p/common_capability.c.o src/libpciaccess.so.0.11.1.p/common_device_name.c.o src/libpciaccess.so.0.11.1.p/common_map.c.o src/libpciaccess.so.0.11.1.p/linux_sysfs.c.o src/libpciaccess.so.0.11.1.p/linux_devmem.c.o src/libpciaccess.so.0.11.1.p/common_vgaarb.c.o -Wl,--as-needed -Wl,--no-undefined -shared -fPIC -Wl,-soname,libpciaccess.so.0 -L/nix/store/k7znyh6bii9bklasnlrn8xlli0rwgvy6-zlib-aarch64-unknown-linux-gnu-1.3.1/lib -lz
aarch64-unknown-linux-gnu-ld: error: unable to find library -lz
```

I believe this is addressed by #307540 (without `--undefined-version` zlib's configure script detects "no shared library support" and doesn't produce `libz.so`, leading to this error in rdeps).

**Comment by RossComputerGuy** [comment]
Yeah, this was fixed by that PR.
